### PR TITLE
Add an opt-in beta npm release channel (version-skew Phase 1)

### LINKED ISSUE #266: Add an opt-in beta npm release channel (version-skew Phase 1)
## Problem

The `kcap` CLI and the Capacitor server share a consolidated version bump, but they're distributed differently. The server rolls out per tenant tier (`internal` first, then `stable`/`micro`); the CLI publishes to npm's single `latest` dist-tag on **every** release. So the moment a release goes out for internal tenants, its matching CLI lands on `latest`, and a `stable`/`micro` user who runs `kcap update` (or `npm i -g @kurrent/kcap`) can end up with a CLI **newer than their server**.

That's a hard failure for the version-sensitive surfaces — daemon features, MCP tools, and agent-invoked CLI commands (`recap`, `errors`, `eval`, …) — each of which talks to its own server endpoint. Core recording (hook forwarding) stays backward-compatible, so the blast radius is those discrete features.

Full design: [`docs/superpowers/specs/2026-07-03-cli-server-version-skew-design.md`](docs/superpowers/specs/2026-07-03-cli-server-version-skew-design.md).

## This issue — Phase 1: opt-in beta release channel

Publish internal-first releases to a `beta` npm dist-tag so `latest` stays matched to the `stable`/`micro` cohort, and give internal testers an opt-in `kcap update --beta`. This is a **noise reducer** that keeps the default update path from dragging users ahead of their server — it is deliberately *not* the correctness mechanism.

## Later phases (separate follow-ups per the design)

- **Phase 2** — server-version cache + proactive per-endpoint guards + a `version` field on `GET /auth/config` (the actual safety mechanism).
- **Phase 3** — reactive backstop messaging.
- **Phase 4** — capability endpoint.

### LINKED ISSUE #268: kcap update reads update_check from the dead legacy CapacitorConfig (ignored for v2 configs)
## Problem

`kcap update`'s passive "update available" hint gate reads the **legacy v1 flat config**, which is effectively dead for real users, so `update_check` is likely ignored for anyone on a v2 (profile) config — i.e. essentially everyone.

`UpdateCommand.PrintUpdateHintIfAvailable()` does:

```csharp
var config = await AppConfig.Load();      // returns CapacitorConfig (legacy v1)
if (config?.UpdateCheck == false) return; // gate
```

But `AppConfig.Load()` (`src/Capacitor.Cli.Core/Config/AppConfig.cs`) deserializes `config.json` **directly** as the legacy flat `CapacitorConfig`. Today's on-disk config is **v2 profile-based** (`ProfileConfig` + `ConfigMigration` v1→v2), whose shape doesn't match `CapacitorConfig`'s top-level fields — so every `CapacitorConfig` field comes back at its default.

Compounding it: the source-generated `JsonSerializerContext` does **not** apply C# member-initializer defaults for properties absent from the JSON. So `UpdateCheck` (declared `= true`) deserializes to `false` when absent. (This is the same quirk `Load()` already works around for `DefaultVisibility` via `?? "org_public"`.)

**Net effect:** for a v2 config, `config?.UpdateCheck` can come back `false` → the update hint is silently suppressed. Meanwhile the *live* per-profile setting `Profile.UpdateCheck` (`src/Capacitor.Cli.Core/Config/ProfileConfig.cs:74`) — the one `kcap config`/`kcap setup` actually writes — is never read by the update path.

## Fix direction

Read `update_check` from the active v2 profile, exactly the pattern #267 established for `update_channel`:

```csharp
var profile = await AppConfig.GetActiveProfileAsync();
if (profile?.UpdateCheck == false) return;
```

## Context

- Surfaced during #267 (beta release channel), which moved `update_channel` to the v2 `Profile` and left this pre-existing `update_check` read on the legacy path. #267's `PrintUpdateHintIfAvailable` is now half-migrated — it reads the *channel* via `GetActiveProfileAsync()` (correct, v2) but still reads the *gate* via `AppConfig.Load()` (v1).
- Worth auditing other `AppConfig.Load()` callers for the same v1/v2 mismatch, and consider whether the legacy `Load()`/`CapacitorConfig` path should be retired now that migration is well established.

### PR DESCRIPTION
**Pull request by alexeyzimarev**
## What & why

The kcap CLI and the Capacitor server share a consolidated version bump but ship through different channels: the server rolls out per tenant tier (`internal` → `stable`/`micro`) while the CLI publishes to npm's single `latest` dist-tag on every release. So an internal-first release drags `stable`/`micro` users onto a CLI newer than their server — a hard failure for daemon features, MCP tools, and agent-invoked commands (core recording stays backward-compatible).

This is **Phase 1** of the [version-skew design](docs/superpowers/specs/2026-07-03-cli-server-version-skew-design.md): an opt-in **beta npm release channel** so internal-first releases go to a `beta` dist-tag and `latest` stays matched to the `stable`/`micro` cohort. It's a noise reducer, **not** the correctness mechanism — that's Phase 2 (server-version guards), tracked separately.

Closes #266
Closes #268
AI-1177

## What's in it

- **Prerelease-aware SemVer 2.0 comparator** (`PrereleaseSemver`) — orders `beta.1 < beta.2 < release`, which the existing `SemverCompare` (strips prerelease) can't.
- **Release workflow** — prerelease git tags publish every package with `npm publish --tag beta`, leaving `latest` untouched; `scripts/npm-dist-tag.sh` derives the tag; `RELEASING.md` documents the invariant (internal-first → prerelease; `latest` only once `stable`/`micro` can serve it).
- **`update_channel` on the v2 `Profile`** (default `latest`).
- **Channel-aware `kcap update --beta`/`--stable`** — persists the choice onto the *resolved* active profile, queries the matching dist-tag, per-channel cache, prerelease-aware comparison, and exposes `install_tag` in the `--check` JSON.
- **Launcher** forwards `--beta`/`--stable` to its probe and installs the resolved dist-tag (instead of hardcoded `@latest`).
- **Docs** — README (Getting started + `kcap update`) and `help-update.txt`.

Default behavior is unchanged: no opt-in ⇒ `latest`.

## Also in this PR: retire the live v1 config read path (Closes #268)

Working on the channel setting surfaced that `kcap update`'s `update_check` gate (and `ImportCommand`) read the **dead legacy flat config** via `AppConfig.Load()`, which returns all-default values for v2-profile users. Rather than patch one caller, this retires the whole live v1 read path:

- Deleted `AppConfig.Load()`; moved `UpdateCommand` and `ImportCommand` onto the v2 profile (`GetActiveProfileAsync()`).
- Demoted `CapacitorConfig` → `LegacyV1Config`, now a **migration-only DTO** used solely by `ConfigMigration`; removed the unused v1 indented JSON context.
- Fixed an STJ source-gen quirk where a v1 config lacking `update_check` migrated to `false` instead of the v1 default `true`.
- **Kept the v1→v2 migration** as the straggler safety net (removing it would silently reset a v1 config on disk).

## Notable during review

- `update_channel` was moved from the legacy v1 `CapacitorConfig` to the v2 `Profile` — the legacy record is dead for real (v2-profile) users and persisting it would corrupt the config.
- Two cross-task bugs caught by the whole-branch review and fixed: the launcher not forwarding the channel flag (feature was a no-op for npm installs), and persistence targeting the wrong profile for multi-profile / `KCAP_PROFILE` / repo-bound users.

## Testing

- 2171 unit tests pass; new coverage: `PrereleaseSemver` precedence, `update_channel` round-trip, `ResolveChannel`, v1→v2 `ConfigMigration` (incl. the `update_check` absent/false/true trio), a WireMock integration test for the channel dist-tag query, and the launcher `resolveInstallSpec`/`probeArgs` (node).
- AOT publish clean (no IL3050/IL2026).

## Follow-ups (not in this PR)

- Polish: a couple of SemVer edge cases (unreachable from our `X.Y.Z-beta.N` scheme), a `--check`+`--beta` doc note, and a binary↔launcher `install_tag` contract test.
- **Phases 2–4** of the design (the actual server-version safety mechanism).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/Capacitor.Cli/Commands/UpdateCommand.cs [hunk 1]
```diff
@@ -87,14 +133,22 @@ public static async Task PrintUpdateHintIfAvailable() {
...
+    /// <summary>
+    /// Per-channel cache path so a <c>beta</c> check doesn't clobber the
+    /// cached <c>latest</c> result (and vice versa).
+    /// </summary>
+    static string CachePathFor(string channel) =>
+        PathHelpers.ConfigPath($"update-check-{channel}.json");
+
+    internal static async Task<(string? latest, string? current)> CheckForUpdateAsync(bool forceCheck, string channel) {
+        var current   = GetCurrentVersion();
+        var cachePath = CachePathFor(channel);
```
[the remark this answers is not in the document]
Comment (alexeyzimarev) [reply src/Capacitor.Cli/Commands/UpdateCommand.cs:150]: Fixed in e5e39ed: `ResolveChannel` now validates the resolved channel against a `{latest, beta}` allowlist (normalized, falling back to `latest`), so an unvalidated `update_channel` from disk can't reach the cache path or registry URL. Added tests for `..`, rooted paths, unknown values, and case normalization. Agree the practical severity is low (it needs write access to the user's own config), but the input-validation + correctness fix is worth it.
